### PR TITLE
Make FabricMegatronMixedPrecision match MegatronMixedPrecision

### PR DESCRIPTION
**Pull request by marcromeyn**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/fabric/plugins.py [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 from nemo.lightning.fabric.conversion import to_fabric
 from nemo.lightning.pytorch.plugins.mixed_precision import (
-    MegatronMixedPrecision, 
     DtypeConfig,
+    MegatronMixedPrecision,
     get_optim_config,
-    update_config_with_dtype_overrides
+    update_config_with_dtype_overrides,
 )
+from nemo.utils.import_utils import safe_import
```
Comment (akoumpa) [inline nemo/lightning/fabric/plugins.py:16]: is this used?

Comment (marcromeyn) [reply nemo/lightning/fabric/plugins.py:16]: Yeah, [here](https://github.com/NVIDIA/NeMo/blob/c1cfa9c0cee7a674e804401a90d2bf4a40afd113/nemo/lightning/fabric/plugins.py#L54).

File: nemo/lightning/fabric/plugins.py [hunk 2]
```diff
@@ -52,64 +108,45 @@ def convert_output(self, data: AnyT) -> AnyT:
...
-        if not hasattr(module, "module"):
-            return module
-
         from megatron.core.transformer.module import Float16Module
         from megatron.core.utils import get_model_config
 
-        if self.precision in ["16-mixed", "bf16-mixed"]:
+        if self.dtype_config.fp16 or self.dtype_config.bf16:
+            # Patch config options
             config = get_model_config(module.module)
```
Comment (cuichenx) [inline nemo/lightning/fabric/plugins.py:130]: should this line be guarded by `if hasattr(module, 'module'):`?

Comment (marcromeyn) [reply nemo/lightning/fabric/plugins.py:130]: Yeah, I realized that as well. The fix is part of this [PR](https://github.com/NVIDIA/NeMo/pull/10879) that was just merged.

**Pull request merged**